jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,854 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link Handler.Wrapper} that conditionally handles a {@link Request}.
+ * The conditions are implemented by {@link IncludeExclude}s of:
+ * <ul>
+ *     <li>A HTTP method name, which can be efficiently matched</li>
+ *     <li>A {@link PathSpec} or string representation, which can be efficiently matched.</li>
+ *     <li>An arbitrary {@link Predicate} taking the {@link Request}, which is matched in a linear test of all predicates.</li>
+ * </ul>
+ *
+ * <p>If the conditions are met, the abstract {@link #onConditionsMet(Request, Response, Callback)} method will be invoked,
+ * otherwise the {@link #onConditionsNotMet(Request, Response, Callback)} method will be invoked.  Implementations may call
+ * the {@link #nextHandler(Request, Response, Callback)} method to call the wrapped handler.</p>
+ *
+ * <p>A typical usage is to extend the {@link Abstract} sub class and provide an implementation of
+ * {@link #onConditionsMet(Request, Response, Callback)} and {@link #onConditionsNotMet(Request, Response, Callback)}:</p>
+ * <pre>{@code
+ * public class MyOptionalHandler extends ConditionalHandler.Abstract
+ * {
+ *     @Override
+ *     public boolean onConditionsMet(Request request, Response response, Callback callback)
+ *     {
+ *         response.getHeaders().add("Test", "My Optional Handling");
+ *         return nextHandle(request, response, callback);
+ *     }
+ *
+ *     @Override
+ *     public boolean onConditionsNoMet(Request request, Response response, Callback callback)
+ *     {
+ *         return false;
+ *     }
+ * }
+ * }</pre>
+ *
+ * <p>If the conditions added to {@code MyOptionalHandler} are met, then the {@link #onConditionsMet(Request, Response, Callback)}
+ * method is called and a response header added before invoking {@link #nextHandler(Request, Response, Callback)}, otherwise
+ * the {@link #onConditionsNotMet(Request, Response, Callback)} is called, which returns false to indicate no more handling.</p>
+ *
+ * <p>Alternatively, one of the concrete subclasses may be used.  These implementations conditionally provide a specific
+ * action in their {@link #onConditionsMet(Request, Response, Callback)} methods. Otherwise, these subclasses are all extension
+ * of the abstract {@link ElseNext} subclass, that implements {@link #onConditionsNotMet(Request, Response, Callback)} to
+ * call {@link #nextHandler(Request, Response, Callback)}.
+ *
+ * <ul>
+ *     <li>{@link DontHandle} - If the conditions are met, terminate further handling by returning {@code false}</li>
+ *     <li>{@link Reject} - If the conditions are met, reject the request with a {@link HttpStatus#FORBIDDEN_403} (or other status code) response.</li>
+ *     <li>{@link SkipNext} - If the conditions are met, then the {@link #getHandler() next handler} is skipped and the
+ *     {@link Singleton#getHandler() following hander} invoked instead.</li>
+ * </ul>
```

Comment:
This seems to be missing the `ElseNext` action

Reply:
The `ElseNext` action was described before the list.  I will move it until after the list.